src/styles/main.scss

```diff
@@ -1,3 +1,70 @@
-body {
-  margin: 0;
+@import "./utils/variables.scss";
+@import "./utils/normalize.scss";
+
+.container {
+  margin-top: 20px;
+  display: flex;
+  justify-content: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: $gap;
+  padding: $table-paddings;
+
+  $days:
+    mon 1,
+    tue 2,
+    wed 3,
+    thu 4,
+    fri 5,
+    sat 6,
+    sun 7;
+
+  @each $day, $day-number in $days {
+    &--start-day--#{$day} {
+      :first-child {
+        margin-left: (($day-number - 1) * $cell-size)
+          + ($gap * ($day-number - 1));
+      }
+    }
+  }
+
+  @for $month-end from 28 through 31 {
+    &--month-length--#{$month-end} {
+      :nth-child(n + #{$month-end}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    height: $cell-size;
+    width: $cell-size;
+    border: 1px $numbers-color solid ;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: $cell-color;
+    transition: background-color 0.5s, transform 0.5s;
+
+    @for $number from 1 through 31 {
+      &:nth-child(#{$number})::before {
+        content: "#{$number}";
+        font-family: $numbers-font;
+        font-size: $numbers-hight;
+        color: $numbers-color;
```

Comment:
There is no need to set styles in a loop, which can be set once for the entire class. This affects performance.